### PR TITLE
metalink: resolve all schemes when None; Fix #1262

### LINKED ISSUE #1262: list_replicas with metalink should resolve all schemes, if no schemes are given
Motivation
----------
Currently the schemes have to be given explicitly. However, if no schemes are given all schemes should be replied, not only the top-priority one.

### PR DESCRIPTION
**Pull request by mlassnig**
metalink: resolve all schemes when None; Fix #1262

### CODE REVIEW

File: lib/rucio/web/rest/webpy/v1/replica.py [hunk 1]
```diff
@@ -102,6 +102,10 @@ def GET(self, scope, name):
             if 'limit' in params:
                 limit = int(params['limit'][0])
 
+        # Resolve all reasonable protocols when doing metalink for maximum access possibilities
+        if metalink and schemes is None:
+            schemes = ['gsiftp', 'srm', 'root', 'davs', 'https', 'file']
```
Comment (bari12) [inline lib/rucio/web/rest/webpy/v1/replica.py]: We have lists of all possible schemes in several places in the code. I wonder if we should replace this with something centrally specified or maybe even something community specific?

Comment (mlassnig) [reply lib/rucio/web/rest/webpy/v1/replica.py]: While I agree a central place makes sense, I don't really see much need for a configurable list as the supported protocols per VO will be automatically resolved through the RSEs anyway?

Comment (bari12) [reply lib/rucio/web/rest/webpy/v1/replica.py]: Yes, I was just putting it in the room. A central list of all rucio supported protocols should be sufficient, the actual VO support is derived from the used protocols anyway.

**Review by bari12 — DISMISSED** [review]
I think the comment got lost 😄 Please change the schemes list to some central defined list (Maybe also in other parts of the code base)

**Comment by bari12** [comment]
Can you please rebase this and re-submit. travis seems to have problems.

**Comment by mlassnig** [comment]
@bari12 done, travis is back in action now

**Review by bari12 — APPROVED** [review]

**Pull request merged**